### ISSUE TITLE
Application lock, when running on second screen.

### ISSUE THREAD

**Issue by Superman2019AAC** [godotengine/godot#92932]
### Tested versions

I found the error in version 4.2. I have tested the beta of 4.3 and it seems that it does not have that error at the moment.

### System information

Windows 11 - Godot v4.2.2.stable - All versions - Nvidia Geforce Asus RTX 4090

### Issue description

When opening the program on a second screen it freezes.

### Steps to reproduce

When I open Godot on the main screen, it works fine. However, when I move it to another monitor and press the plus button to add a new node, it freezes. I have tested Godot version v4.3 beta1 and this error does not occur.

On some occasions, Godot runs on the main screen, but when adding a node, the node window opens as a floating window on the second monitor. This causes Godot to freeze and close.

### Minimal reproduction project (MRP)

Any project

**Comment by AThousandShips** [comment]
Since it doesn't occur in 4.3 it might have been fixed directly, if so it might have been cherry picked, if it hasn't it'd be good to identify what fix solved it, and try cherry picking it

Could you try older development releases of 4.3? Going backwards with dev6, dev5, etc., it would help a lot to identify in what range the stops occurring in 4.3!

**Comment by Superman2019AAC** [comment]
I have done more testing. The beta version also has the same error. It ONLY occurs with the Compatible rendering model. It does not happen in the Forward Plus and Mobile modes.

**Comment by AThousandShips** [comment]
So it happens on the compatibility method on both 4.2.2 and 4.3.beta1? Just to clarify

**Comment by Superman2019AAC** [comment]
That's right. It happens in 4.3 beta, 4.2.2 stable, and 4.2.1 stable, which are the versions I've tested. I thought it didn't happen in 4.3 because I started a new project with Forward Plus and it worked correctly. It must be a bug or issue with the Compatible mode.
